### ISSUE TITLE
Could you add a license?

### ISSUE THREAD

**Issue by Lanius-collaris** [cubing/AnimCubeJS#20]

**Comment by bcube2** [comment]
The MIT license has been added.

**Comment by lgarron** [comment]
> The MIT license has been added.

I think we need to be a bit careful here. At least some of the code in the repository is GWT compiled but originally from Josef Jelinek. Would you mind breaking down all the source code authors to check what can be published under which license?

**Comment by bcube2** [comment]
"_Would you mind breaking down all the source code authors to check what can be published under which license?_"

As far as I know, there are 2 source code authors: Josef for AnimCube (Java) and Michael for AnimCubeJS (both the old version (GWT and Apache Ant compiled) and the current version (Vanilla JS)). I have checked with both of them that the MIT license as presented is OK for them. Tadeáš mentioned in the license is a source code co-author for the documentation web pages and he is also OK with the MIT license.

I have also checked with Michael that he is OK with adding no license for the old version of AnimCubeJS.

If you see any issue with the MIT license feel free to tell me.

Edit: even though AnimCubeJS is a standalone piece of software, for demonstration / documentation purposes also a third-party software (all of which is optional (i.e. not needed for users to use) and under the MIT license) can be found in the repository. One exception regarding licensing is Bruce´s 2x2x2 Hamiltonian circuit move sequence. I was asking Bruce for a permission to distribute his move sequence before publishing it in the form which can be found in the documentation web page.

There aren´t any other source code authors associated with AnimCubeJS I am aware of.

**Comment by lgarron** [comment]
Thanks, that's all very helpful to have in writing. :-D

Could I ask for two followups?

- Could you get documentation like screenshots or statements from the mentioned authors? I know that's paranoid, but I want to be really careful that we can back up any statements for licenses in the `cubing` org (or abandon old projects like JSSS that cannot easily be licensed).
- If `old` isn't licensed, could we move it into another branch, or at least add a `LICENSE.md` inside that folder clarifying that the current project license doesn't apply to its contents?

**Comment by bcube2** [comment]
"_Could you get documentation like screenshots or statements from the mentioned authors?_"

No problem. However, since my discussions with Michael, Josef and Bruce were all in private, I am going to respect their privacy and send you a private e-mail in which the e-mail addresses being not publicly available are censored and some private discussions are shortened. Thank you for your understanding.

If that´s not good enough for you, I will think about a more clear way as proposed here: https://github.com/cubing/cubing.js/issues/208. But hopefully it will be good enough for you :-) 

"_If `old` isn't licensed, could we move it into another branch, or at least add a `LICENSE.md` inside that folder clarifying that the current project license doesn't apply to its contents?_"

I am fine with either way :-). Allow me to clarify my previous statement "I have also checked with Michael that he is OK with adding no license for the old version of AnimCubeJS.". As long as the MIT license is also applicable for the `old` folder, I have checked with Michael he is OK with adding the MIT license to one place in the repository (thus covering both versions of AnimCubeJS). As long as the MIT license is not applicable for the `old` folder, we would rather keep the `old` folder without a license than adding an extra license for it.

All source codes available in the `old` folder are from Michael, Josef or Tadeáš, except for the optional blazy.js script placed here: https://github.com/cubing/AnimCubeJS/tree/gh-pages/old/sources
